plugin/src/main/cpp/include/extensions/openxr_meta_performance_metrics_extension_wrapper.h

```diff
@@ -0,0 +1,137 @@
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#pragma once
+
+#include <openxr/openxr.h>
+#include <godot_cpp/classes/open_xr_extension_wrapper_extension.hpp>
+#include <godot_cpp/variant/packed_string_array.hpp>
+#include <map>
+
+#include "util.h"
+
+using namespace godot;
+
+class OpenXRMetaPerformanceMetricsExtensionWrapper : public OpenXRExtensionWrapperExtension {
+	GDCLASS(OpenXRMetaPerformanceMetricsExtensionWrapper, OpenXRExtensionWrapperExtension);
+
+public:
+	enum PerformanceMetricsCounterFlags {
+		PERFORMANCE_METRICS_COUNTER_FLAGS_ANY_VALUE_VALID_BIT = 1,
+		PERFORMANCE_METRICS_COUNTER_FLAGS_UINT_VALUE_VALID_BIT = 2,
+		PERFORMANCE_METRICS_COUNTER_FLAGS_FLOAT_VALUE_VALID_BIT = 4,
+	};
+
+	enum PerformanceMetricsCounterUnit {
+		PERFORMANCE_METRICS_COUNTER_UNIT_GENERIC,
+		PERFORMANCE_METRICS_COUNTER_UNIT_PERCENTAGE,
+		PERFORMANCE_METRICS_COUNTER_UNIT_MILLISECONDS,
+		PERFORMANCE_METRICS_COUNTER_UNIT_BYTES,
+		PERFORMANCE_METRICS_COUNTER_UNIT_HERTZ,
+	};
+
+	static OpenXRMetaPerformanceMetricsExtensionWrapper *get_singleton();
+
+	OpenXRMetaPerformanceMetricsExtensionWrapper();
+	~OpenXRMetaPerformanceMetricsExtensionWrapper();
+
+	godot::Dictionary _get_requested_extensions() override;
+
+	void _on_instance_created(uint64_t instance) override;
+	void _on_instance_destroyed() override;
+	void _on_state_ready() override;
+
+	bool is_enabled() { return meta_performance_metrics_ext; }
+
+	void set_performance_metrics_enabled(bool p_enabled);
+	bool is_performance_metrics_enabled();
```

Comment:
Again, maybe the we could use the word "capture" rather than the more generic "enable"?

Maybe:

```suggestion
	void set_capture_performance_metrics(bool p_enabled);
	bool is_capturing_performance_metrics();
```

Reply:
Applied!